### ISSUE TITLE
Missing `[compile]` and `[link]` annotations in some settings

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#18203]
Some settings in `settings.js` have neither `[compile]`, `[link]` or `[compile+link]` annotations. These are particularly useful for https://emsettings.surma.technology/, so decided to report them for tracking:

- [ ] `SUPPORT_BIG_ENDIAN`
- [ ] `GL_EXPLICIT_UNIFORM_LOCATION`
- [ ] `GL_EXPLICIT_UNIFORM_BINDING`
- [ ] `GL_WORKAROUND_SAFARI_GETCONTEXT_BUG`
- [ ] `EXPORT_EXCEPTION_HANDLING_HELPERS`
- [ ] `EXTRA_EXPORTED_RUNTIME_METHODS`
- [ ] `IN_TEST_HARNESS`
- [ ] `LOAD_SOURCE_MAP`
- [ ] `WASM2C_SANDBOXING`
- [ ] `AUTOLOAD_DYLIBS`
- [ ] `ALLOW_UNIMPLEMENTED_SYSCALLS`
- [ ] `POLYFILL`
- [ ] `LEGACY_RUNTIME`
- [ ] `LEGACY_SETTINGS`

cc @surma @sbc100

**Comment by sbc100** [comment]
I think we can try to make this automatic somehow since we know precisely which settings are allowed at compile time: https://github.com/emscripten-core/emscripten/blob/da8c8f8992f001eae65a56573401dbed10c9fb4c/tools/settings.py#L55-L57

**Comment by RReverser** [comment]
Oh that would be great. Since that Python already "parses" the settings.js, I suppose it would be easy to use it to overwrite settings.js with updated annotations if any in a separate script? Then it's a matter of adding it as a pre-commit hook or something.

**Comment by sbc100** [comment]
Yes, it would be cool to move the source of truth to those comments.

Even better we could switch the format of the settings file to `json with comments` ... but parsing `json with comments` isn't something python can do by default.

(BTW i've been wanting to convert the format from JS for a while now since we don't actually ever parse this file as JS anymore!)

**Comment by RReverser** [comment]
> switch the format of the settings file to `json with comments` ... but parsing `json with comments` isn't something python can do by default.

Oh yeah you could use JSON5 for that. https://pypi.org/project/json5/

**Comment by sbc100** [comment]
Yeah, sadly we got some pushback last time we tried to start adding 3rd party python modules..  I might try again.

**Comment by RReverser** [comment]
Ugh. I mean, you could also pre-process the input yourself and remove line comments with a `//.*` regex for now...
